diff --git a/etc/di.php b/etc/di.php
index f5b1f57e..acb86b2a 100644
--- a/etc/di.php
+++ b/etc/di.php
@@ -7,6 +7,7 @@ return [
     "Starbug\Core" => "core/app",
     "Starbug\Log" => "modules/log",
     "Starbug\State" => "modules/state",
+    "Starbug\EventDispatcher" => "modules/event-dispatcher",
     "Starbug\Db" => "modules/db",
     "Starbug\Doctrine" => "modules/doctrine",
     "Starbug\ExternalDb" => "app/modules/external-db",
@@ -18,6 +19,7 @@ return [
     "Starbug\Js" => "modules/js",
     "Starbug\Tachyons" => "modules/tachyons",
     "Starbug\Intl" => "modules/intl",
+    "Starbug\Imports" => "modules/imports",
     "Starbug\Content" => "modules/content",
     "Starbug\Salesforce" => "app/modules/salesforce",
     "Starbug\Saml" => "app/modules/saml",
